config.json

```diff
@@ -1412,6 +1412,20 @@
             "integers"
         ]
     },
+    {
+      "slug": "retree",
+      "uuid": "18467e72-e3b0-403c-896a-ce337f96e6f7",
+      "core": false,
+      "unlocked_by": "markdown",
+      "difficulty": 7,
+      "topics": [
+        "functional_programming",
+        "pattern_recognition",
+        "recursion",
+        "transforming",
+        "trees"
```

Comment:
I'm not sure `functional_programming` applies, as there is nothing about the exercise that *requires* functional programming.

Same with `transforming`, but if you can make a solid case for it, I'm fine with keeping it.

`recursion` is fine; I attempted an iterative solution to this, and it is not trivial, so `recursion` is a good hint towards a solution.

Reply:
1. will remove functional_programming, as I thought I already did when I saw recursion was an option.
2.  I thought "transforming" was appropriate, as you are literally transforming two lists into an object representing a tree (a nested dict in this case).
3.  You can do this iteratively, but, IIRC, that requires a proper setup with a stack and a deque, and I find recursive makes more sense on a first pass if you aren't running the risk of hitting your recursion limit.